### ISSUE TITLE
Error on installation

### ISSUE THREAD

**Issue by fpetitit** [typicode/hotel#1]
Hello,
when trying to install Hotel using the command "sudo npm install -g hotel", I get this error : 
>/usr/local/bin/hotel -> /usr/local/lib/node_modules/hotel/bin/index.js
> hotel@0.2.1 install /usr/local/lib/node_modules/hotel
> node bin/install
> /usr/local/lib/node_modules/hotel/node_modules/mkdirp/index.js:90
>                    throw err0;
>                         ^
>Error: EACCES, permission denied '/Users/my_user_name/.hotel'
>[...]

I tried to uninstall multiple times but always get the same error :/

**Comment by typicode** [comment]
Hi,

Actually, I've not tried installing using `sudo` (the project doesn't require admin privileges btw). It may creates problems because `~/.hotel` is owned by `root`.

To uninstall "manually", you can try `sudo npm rm -g hotel -f && rm -rf ~/.hotel ~/Library/LaunchAgents/hotel.plist`

Then, you can retry using `npm install -g hotel`.

I'll add https://github.com/sindresorhus/downgrade-root and see if it helps.

**Comment by typicode** [comment]
Just released `v0.2.2` with `downgrade-root`. It should work with `sudo`.

**Comment by fpetitit** [comment]
Thank you !
I just managed to go further by not using "sudo" to install global package, but I know obtain the following error, using Hotel v0.2.2 : 

 Create  ~/Library/LaunchAgents/hotel.plist

/Users/fpe/npm/lib/node_modules/hotel/lib/actions/autostart.js:45
    cp.execSync('launchctl load -Fw ' + file[platform]);
       ^
TypeError: Object #<Object> has no method 'execSync'
    at Object.create (/Users/fpe/npm/lib/node_modules/hotel/lib/actions/autostart.js:45:8)
    at Object.<anonymous> (/Users/fpe/npm/lib/node_modules/hotel/bin/install.js:14:11)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:935:3

npm ERR! hotel@0.2.2 install: `node bin/install`
npm ERR! Exit status 8
npm ERR!
npm ERR! Failed at the hotel@0.2.2 install script.
npm ERR! This is most likely a problem with the hotel package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node bin/install
npm ERR! You can get their info via:
npm ERR!     npm owner ls hotel
npm ERR! There is likely additional logging output above.
npm ERR! System Darwin 14.3.0
npm ERR! command "node" "/usr/local/bin/npm" "install" "-g" "hotel"
npm ERR! cwd /Users/fpe
npm ERR! node -v v0.10.38
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE

> hotel@0.2.2 uninstall /Users/fpe/npm/lib/node_modules/hotel
> node bin/uninstall


  Remove  ~/Library/LaunchAgents/hotel.plist

/Users/fpe/npm/lib/node_modules/hotel/lib/actions/autostart.js:58
    cp.execSync('launchctl unload ' + file[platform]);
       ^
TypeError: Object #<Object> has no method 'execSync'
    at Object.remove (/Users/fpe/npm/lib/node_modules/hotel/lib/actions/autostart.js:58:8)
    at Object.<anonymous> (/Users/fpe/npm/lib/node_modules/hotel/bin/uninstall.js:5:11)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Function.Module.runMain (module.js:497:10)
    at startup (node.js:119:16)
    at node.js:935:3
npm WARN continuing anyway hotel@0.2.2 uninstall: `node bin/uninstall`
npm WARN continuing anyway Exit status 8
npm ERR! not ok code 0

Perhaps a problem with my version of NodeJS (0.10.38) ?

**Comment by typicode** [comment]
Yes, you're right. I've just added an `engine` property to `package.json` so that people get a warning when trying to install with node < 0.12.

**Comment by syzer** [comment]
i also cannot `sudo npm install -g hotel` on mac,
first had no acces rights to `~/.hotel/` so i made it with `777` rights
but then:
```
15406 verbose stack Error: hotel@0.2.7 install: `node bin/install`
15406 verbose stack Exit status 1
15406 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:213:16)
15406 verbose stack     at EventEmitter.emit (events.js:110:17)
15406 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
15406 verbose stack     at ChildProcess.emit (events.js:110:17)
15406 verbose stack     at maybeClose (child_process.js:1015:16)
15406 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
15410 error argv "node" "/usr/local/bin/npm" "install" "-g" "hotel"
15411 error node v0.12.4
15412 error npm  v2.10.1
15413 error code ELIFECYCLE
15414 error hotel@0.2.7 install: `node bin/install`
15414 error Exit status 1
15415 error Failed at the hotel@0.2.7 install script 'node bin/install'.
15415 error This is most likely a problem with the hotel package,
15415 error not with npm itself.
15415 error Tell the author that this fails on your system:
15415 error     node bin/install
15415 error You can get their info via:
15415 error     npm owner ls hotel
15415 error There is likely additional logging output above.
15416 verbose exit [ 1, true ]
15417 verbose unbuild lib/node_modules/hotel
15418 info preuninstall hotel@0.2.7
15419 info uninstall hotel@0.2.7
15420 verbose unsafe-perm in lifecycle false
15421 verbose unbuild rmStuff hotel@0.2.7 from /usr/local/lib/node_modules
15422 silly gentlyRm /usr/local/bin/hotel is being gently removed
15423 silly gentlyRm verifying /usr/local is an npm working directory
15424 silly gentlyRm containing path /usr/local is under npm's control, in /usr/local
15425 silly gentlyRm deletion target /usr/local/bin/hotel is under /usr/local
15426 verbose gentlyRm vacuuming from /usr/local/bin/hotel up to /usr/local
15427 silly vacuum-fs removing /usr/local/bin/hotel
15428 silly vacuum-fs quitting because other entries in /usr/local/bin
15429 info postuninstall hotel@0.2.7
15430 silly gentlyRm /usr/local/lib/node_modules/hotel is being purged from base /usr/local
15431 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/hotel
15432 silly vacuum-fs purging /usr/local/lib/node_modules/hotel
15433 silly vacuum-fs quitting because other entries in /usr/local/lib/node_modules
```

**Comment by typicode** [comment]
Thanks for the feedback @syzer

Just released `v0.2.8` and updated `README.md` explaining how to install with `sudo`.

Also if the command is being installed using `sudo`, you'll get a message in the terminal telling you to run `hotel autostart && hotel start` to complete installation.

This way nothing will be created with root permissions.

Just in case, to start fresh, please run `rm -rf ~/.hotel && ~/Library/LaunchAgents/hotel.plist`

**Comment by syzer** [comment]
:+1: awesome
